### PR TITLE
update frontend and mk_ck_lib

### PR DESCRIPTION
**Pull request by fsx950223**

### CODE REVIEW

**Comment by fsx950223** [comment]
@chenyang78 @ipiszy review this one too.

File: python/aitemplate/frontend/nn/attention.py [hunk 1]
```diff
@@ -361,32 +362,55 @@ def __init__(
...
+        OP = ops.bmm_softmax_bmm_permute(
+            shape=(self.num_heads,),
+            scale=head_dim**-0.5,
+            causal=self.causal,
         )
-        return self.op(query, key, value)
+        return OP(query, key, value)
 
-    def forward(self, *args):
+    def forward(self, *args, seqlens=None):
```
Comment (ipiszy) [inline python/aitemplate/frontend/nn/attention.py:408]: What does seqlen represent? Where is seqlens used?

File: python/aitemplate/frontend/nn/attention.py [hunk 2]
```diff
@@ -395,7 +419,8 @@ def forward(self, *args):
         else:
             x = self.proj(attn_output)
         x = self.proj_drop(x)
-        x = ops.reshape()(x, [batch, -1, self.dim])
+        if not isinstance(batch, IntVar):
+            x = ops.reshape()(x, [batch, -1, self.dim])
```
Comment (ipiszy) [inline python/aitemplate/frontend/nn/attention.py:423]: This looks weird. Why this change?

Comment (fsx950223) [reply python/aitemplate/frontend/nn/attention.py:423]: ROCM attention support [batch * seqlen, nheads] tensor as inputs and output [batch * seqlen, nheads] tensor. The branch keep original behavior.

File: python/aitemplate/frontend/nn/attention.py (hunk 1 above)
Comment (fsx950223) [reply python/aitemplate/frontend/nn/attention.py:408]: It's used in rocm attention.

Comment (ipiszy) [reply python/aitemplate/frontend/nn/attention.py:408]: I feel the API after this change is not very readable. This API is supposed to be backend-agnostic, and is shared between CUDA and ROCM. Can we avoid this change?

Comment (fsx950223) [reply python/aitemplate/frontend/nn/attention.py:408]: rocm attention backend needs seqlen.

**Comment by fsx950223** [comment]
I have reverted the attention
